keps/sig-node/4815-dra-partitionable-devices/README.md

```diff
@@ -0,0 +1,2140 @@
+        quantity: "0"
+- name: common-mig-1g.5gb-me-nvidia-a100-sxm4-40gb
+  composite:
+    attributes:
+      profile:
+        string: 1g.5gb+me
+    capacity:
+      copy-engines:
+        quantity: "1"
+      decoders:
+        quantity: "1"
+      encoders:
+        quantity: "0"
+      jpeg-engines:
+        quantity: "1"
+      memory:
+        quantity: 4864Mi
+      multiprocessors:
+        quantity: "14"
+      ofa-engines:
+        quantity: "1"
+- name: common-mig-1g.5gb-nvidia-a100-sxm4-40gb
+  composite:
+    attributes:
+      profile:
+        string: 1g.5gb
+    capacity:
+      copy-engines:
+        quantity: "1"
+      decoders:
+        quantity: "0"
+      encoders:
+        quantity: "0"
+      jpeg-engines:
+        quantity: "0"
+      memory:
+        quantity: 4864Mi
+      multiprocessors:
+        quantity: "14"
+      ofa-engines:
+        quantity: "0"
+- name: common-mig-2g.10gb-nvidia-a100-sxm4-40gb
+  composite:
+    attributes:
+      profile:
+        string: 2g.10gb
+    capacity:
+      copy-engines:
+        quantity: "2"
+      decoders:
+        quantity: "1"
+      encoders:
+        quantity: "0"
+      jpeg-engines:
+        quantity: "0"
+      memory:
+        quantity: 9984Mi
+      multiprocessors:
+        quantity: "28"
+      ofa-engines:
+        quantity: "0"
+- name: common-mig-3g.20gb-nvidia-a100-sxm4-40gb
+  composite:
+    attributes:
+      profile:
+        string: 3g.20gb
+    capacity:
+      copy-engines:
+        quantity: "3"
+      decoders:
+        quantity: "2"
+      encoders:
+        quantity: "0"
+      jpeg-engines:
+        quantity: "0"
+      memory:
+        quantity: 20096Mi
+      multiprocessors:
+        quantity: "42"
+      ofa-engines:
+        quantity: "0"
+- name: common-mig-4g.20gb-nvidia-a100-sxm4-40gb
+  composite:
+    attributes:
+      profile:
+        string: 4g.20gb
+    capacity:
+      copy-engines:
+        quantity: "4"
+      decoders:
+        quantity: "2"
+      encoders:
+        quantity: "0"
+      jpeg-engines:
+        quantity: "0"
+      memory:
+        quantity: 20096Mi
+      multiprocessors:
+        quantity: "56"
+      ofa-engines:
+        quantity: "0"
+- name: common-mig-7g.40gb-nvidia-a100-sxm4-40gb
+  composite:
+    attributes:
+      profile:
+        string: 7g.40gb
+    capacity:
+      copy-engines:
+        quantity: "7"
+      decoders:
+        quantity: "5"
+      encoders:
+        quantity: "0"
+      jpeg-engines:
+        quantity: "1"
+      memory:
+        quantity: 40320Mi
+      multiprocessors:
+        quantity: "98"
+      ofa-engines:
+        quantity: "1"
+- name: common-mig-nvidia-a100-sxm4-40gb-attributes
+  composite:
+    attributes:
+      architecture:
+        string: Ampere
+      brand:
+        string: Nvidia
+      cudaComputeCapability:
+        string: "8.0"
+      productName:
+        string: NVIDIA A100-SXM4-40GB
+      type:
+        string: mig
+- name: memory-slices-0
```

Comment:
All of this would be unnecessary if we had an `intRange` or a `set` abstraction to work with. It only looks as nasty as it does because we need some way of encoding an explicit "set"  of resources, but all we have at our disposable is the ability to define a "quantity" of resources.

For example, this mixin:
```
- name: memory-slices-0-3
  composite:
    capacity:
      memorySlice0:
        quantity: "1"
      memorySlice1:
        quantity: "1"
      memorySlice2:
        quantity: "1"
      memorySlice3:
        quantity: "1"
```
would turn into a specific `capacity` entry on the device itself of something like:
```
capacity:
  memorySlices: [0,1,2,3] // if we supported sets
```
or
```
capacity:
  memorySlices: "0-3" // if we supported intRanges
```

There would be no need for this long list of mixins defining valid memorySlice combinations at all.

Given that this is what we have to work with for now, I think it's acceptable, but we should look into writing a future KEP that introduces `set` and/or `intRange` as a usable abstraction here.